src/LibraryViewExtensionMSWebBrowser/web/library/library.html

```diff
@@ -0,0 +1,136 @@
+﻿<!DOCTYPE html>
+<html>
+
+<head>
+    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <title>Librarie.js sample page</title>
+    <style>
+        body {
+            padding: 0;
+            margin: 0;
+            background-color: #353535;
+        }
+    </style>
+</head>
+
+<body>
+    <!-- Placeholders must exist before all other scripts that try to access them -->
+    <div class="OuterMostContainer" id="libraryContainerPlaceholder"></div>
+    <!-- The main library view component -->
+    <script>
+        LIBPLACEHOLDER
+    </script>
+
+    <script>
+        var replaceImageDelayTime = 100;
+        var currentImage;
+
+        function replaceImages() {
+            var allimages = document.getElementsByTagName("img");
+
+            allimages.forEach(function (element) {
+                currentImage = element;
+                var src = element.src
+                if (element.orgSrc != null) {
+                    src = element.orgSrc;
+                }
+                //the icon is already set - bail.
+                if (src.startsWith("data:image")) {
+                    return;
+                }
+                //request the icon from the extension.
+                window.external.getBase64StringFromPath(src);
```

Comment:
window.external is the `ObjectForScripting` we exposed before - so this lets us call c# functions.